src/shell/lisa_gen_buildroot_rootfs

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+set -eu
+
+BUILDROOT_URI=git://git.busybox.net/buildroot
+BUILDROOT_VERSION=2018.08
+BUILDROOT_DIR=$LISA_HOME/src/buildroot
+BUILDROOT_PATCHES_DIR=$(dirname $0)/buildroot_patches
+BUILDROOT_CONFIG=$(dirname $0)/buildroot_config
+
+function print_usage {
+	echo "Usage: $0 [options]"
+	echo "	options:"
+	echo "		-p: purge buildroot to force a fresh build"
+	echo "		-h: print this help message"
+}
+
+while getopts "hp" opt
+do
+	case $opt in
+		h)
+			print_usage
+			exit 0
+			;;
+		p)
+			rm -rf $BUILDROOT_DIR
+			exit 0
+			;;
+		?)
+			print_usage
+			exit -1
+			;;
+	esac
+done
+
+# Executed command @$1 if file @$2 doesn't exist
+function do_if {
+	if [ ! -e $2 ]; then
+		$1
```

Comment:
Without quotes, bash will perform some expansions a second time, which could be confusing

Reply:
How could the expansion happen for a second time?

Reply:
A first expansion will take place when calling `do_if`, and the second one will take place inside `do_if` when evaluating $1: If you pass the single-quoted parameter `'echo *'` to `do_if`, it should fail since that is not a command name, but in our case it will actually go and expand it. If we intend to support parameters on the function, swapping the file and the command would be a better idea instead of relying on this double expansion:
```
 function do_if {
# usage: do_if FILENAME CMD [[[PARAM1] PARAM2] .....]]
	if [ ! -e "$1" ]; then
                shift
		"$@"
```

Reply:
I've enhanced the description of the function in the newer version

```
# Execute function @$1 once
function do_once {
	FILE="$BUILDROOT_DIR/.lisa_$1"
	if [ ! -e "$FILE" ]; then
		$1
		touch "$FILE"
	fi
}
```

it's not meant to be a generic function and it's meant as a little wrapper to reduce the clutter of repetitive code. We can completely remove it if it's confusing. I didn't quote $1 in the newer version as I couldn't see how double expansion could happen from passing the function name. Don't mind to add the quotes to future proof it.

I would prefer not to over generalize the function though.